tests/src/junit/java/net/neoforged/neoforge/unittest/transfer/InstancedHandlerTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.unittest.transfer;
+
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.material.Fluids;
+import net.neoforged.neoforge.transfer.ResourceHandlerUtil;
+import net.neoforged.neoforge.transfer.handlers.templates.EmptyResourceHandler;
+import net.neoforged.neoforge.transfer.handlers.templates.InfiniteResourceHandler;
+import net.neoforged.neoforge.transfer.handlers.templates.VoidResourceHandler;
+import net.neoforged.neoforge.transfer.resources.FluidResource;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.resources.ItemResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionManager;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.ThrowableAssert;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class InstancedHandlerTests {
+    @Test
+    void emptyHandlers() {
+        //EMPTY no operation handlers but should throw on index calls
+        testEmptyHandler(EmptyResourceHandler.instance(), ItemResource.EMPTY);
+        testEmptyHandler(EmptyResourceHandler.instance(), FluidResource.EMPTY);
+    }
+
+    @Test
+    public void voidHandlers() {
```

Comment:
This doesn't need to be explicitly declared as `public`. Also `testVoidHandlers` or use a display name